File: 2000/4/7-4.py
#!/usr/bin/env python

import sys

cnt = 1
for line in sys.stdin:
    if line == "":
        break
    if line[-1] == "\n":
        lin = line[:-1]
    sentence = line.split()
    for i in xrange(len(sentence)):
        word = sentence.pop(0).lower()
        if word[-1] > "z" or word[-1] < "a":
            word = word[:-1]
        if word != "":
            sentence.append(word)
    sorted = sentence[:]
    sorted.sort()
    reversed = sorted[:]
    reversed.reverse()
    print "Sentence", cnt, "is",
    if sorted != sentence and reversed != sentence:
        print "non-monotonic."
    else:
        print "monotonic."
    cnt += 1

File: 2000/4/7-4.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    string line, word;
    vector<string> sentence, sorted, reversed;
    int cnt = 1;
    while (getline(cin, line))
    {
        cout << "Sentence " << cnt++ << " is ";
        for (int i = 0; i < line.length(); i++)
        {
            switch(line[i])
            {
                case ' ': case '.': case ',': case '-': case '\n':
                    if (!word.empty())
                        sentence.push_back(word);
                    word.clear();
                    break;
                default:
                    word.push_back(tolower(line[i]));
            }
        }
        sorted = sentence;
        sort(sorted.begin(), sorted.end());
        reversed = sorted;
        reverse(reversed.begin(), reversed.end());
        if ((sorted != sentence) && (reversed != sentence))
            cout << "non-";
        cout << "monotonic." << endl;
        sentence.clear();
        word.clear();
        line.clear();
    }
}
